static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx

```diff
@@ -238,6 +246,44 @@ function SelectProjectStep({
   );
 }
 
+function usePublicDSN({
+  organization,
+  project,
+}: {
+  organization: Organization | null | undefined;
+  project: Project | null | undefined;
+}) {
+  const api = useApi();
+  const [response, setResponse] = useState<RequestState<string | null>>({
+    type: 'initial',
+  });
+  useEffect(() => {
+    if (!organization || !project) {
+      return;
+    }
+    setResponse(v => ({...v, type: 'loading', data: null}));
+    const request: Promise<ProjectKey[]> = api.requestPromise(
```

Comment:
Is this set as a const so that we can set the Promise<ProjectKey[]> type? If so, this would be good feedback for core team to expose a generic type on requestpromise 🤔

Reply:
yeah it is. i tried to convert it into a generic and broke a bunch of stuff 😅 so i opted just to cast.

Reply:
i can just cut a separate PR for this and fix the types in the problematic files, its just a handful